addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -0,0 +1,68 @@
+odoo.define('website_sale_stock.website_sale', function(require) {
+'use strict';
+
+var base = require('web_editor.base');
+var ajax = require('web.ajax');
+var core = require('web.core');
+
+var QWeb = core.qweb;
+
+if(!$('.oe_website_sale').length) {
+    return $.Deferred().reject("DOM doesn't contain '.oe_website_sale'");
+}
+
+$('.oe_website_sale').each(function() {
+    var oe_website_sale = this;
+
+    // For options products
+    $(oe_website_sale).on('change', 'input[name="add_qty"]', function() {
+        $(oe_website_sale).find('ul[data-attribute_value_ids]').trigger('change');
+    });
+
+    $(oe_website_sale).find('input[name="add_qty"]').trigger('change');
+
+    // Handle case when manually write in input
+    $(oe_website_sale).on('change', '.js_quantity', function(event) {
+        var $input = $(event.currentTarget);
+        var max_qty = parseInt($input.data('max'));
+        if($input.val() > max_qty) {
+            $input.val(max_qty);
+        }
+    });
+
+    $(oe_website_sale).on('change', 'input.js_variant_change, select.js_variant_change, ul[data-attribute_value_ids]', function(event) {
```

Comment:
it would be nice to add some docstrings to explain what's happening here. It's not very easy to read & understand.